SonarQube.MSBuild.Tasks/WriteProjectInfoFile.cs

```diff
@@ -60,12 +74,12 @@ public override bool Execute()
             pi.ProjectName = this.ProjectName;
             pi.FullPath = this.FullProjectPath;
 
-            // TODO: handle failures and missing values.
             Guid projectId;
             if (Guid.TryParse(this.ProjectGuid, out projectId))
             {
                 pi.ProjectGuid = projectId;
                 pi.AnalysisResults = TryCreateAnalysisResults(this.AnalysisResults);
+                pi.AnalysisSettings = TryCreateAnalysisSettings(this.AnalysisSettings);
```

Comment:
TryCreateAnalysisSettings and TryCreateAnalysisResults do not follow the usual bool TryFoo(out result) pattern